### PR TITLE
Add support for using cadvisor Prometheus endpoint

### PR DESCRIPTION
**Pull request by mnorbury**
#### What does this PR do?

Adds support for using the cadvisor Prometheus endpoint (cProm)

The [testHTTPConn](https://github.com/cloudability/metrics-agent/compare/mnorbury-cadvisor-endpoint?expand=1#diff-7c1ca396f7b3ef69463ad472cf0120e4a854574ec4d837eea82655a8742a300dR385-R405) method has been extended to check the cProm endpoint. Either the cProm endpoint or the stats/container endpoint are required for this method to return `true` - indicating the required endpoints are present.

In addition, the [testHTTPConn](https://github.com/cloudability/metrics-agent/compare/mnorbury-cadvisor-endpoint?expand=1#diff-7c1ca396f7b3ef69463ad472cf0120e4a854574ec4d837eea82655a8742a300dR385-R405) method uses an [EndpointMask](https://github.com/cloudability/metrics-agent/compare/mnorbury-cadvisor-endpoint?expand=1#diff-7c1ca396f7b3ef69463ad472cf0120e4a854574ec4d837eea82655a8742a300dR89-R121) to record the state of the endpoints, to allow the metrics polling loop to only query the available endpoints (e.g. see [here](https://github.com/cloudability/metrics-agent/compare/mnorbury-cadvisor-endpoint?expand=1#diff-7c1ca396f7b3ef69463ad472cf0120e4a854574ec4d837eea82655a8742a300dR300-R306))

If both the cProm endpoint and the stats/containers endpoint are available we will collect both metrics.

#### Where should the reviewer start?

[testHTTPConn](https://github.com/cloudability/metrics-agent/compare/mnorbury-cadvisor-endpoint?expand=1#diff-7c1ca396f7b3ef69463ad472cf0120e4a854574ec4d837eea82655a8742a300dR385-R405)

#### How should this be manually tested?

Tested by deploying to a k8s v1.18 AKS cluster where the stats/container endpoint has been removed. Tested by deploying to a k8s v1.17 to confirm backwards compatibility.

#### Any background context you want to provide?

The stats/containers endpoint is being deprecated and the cProm endpoint provides an alternative source for the same metrics.

#### Developer Done List

- [ ] Tests Added/Updated
- [ ] Updated README.md
- [ ] Verified backward compatible
- [ ] Verified database migrations will not be catastrophic
- [ ] Considered Security, Availability and Confidentiality

#### Reviewer/Approval Done List

- [ ] Tests Pass Locally
- [ ] CI Build Passes
- [ ] Verified README.md is updated
- [ ] Verified changes are backward compatible
- [ ] Reviewed impact to Security, Availability and Confidentiality (if issue found, add comments and request changes)

### CODE REVIEW

File: kubernetes/nodecollection_test.go [hunk 1]
```diff
@@ -71,13 +71,15 @@ func TestEnsureNodeSource(t *testing.T) {
 	// The final 400 fails the direct connection test, and as no proxy client is provided
 	// for the "unsuccessful" node test, it falls through to unreachable status.
 	// If no status code is provided, the default response is 200.
-	returnCodes := []int{200, 200, 400, 400, 200, 200, 400}
+	returnCodes := []int{200, 200, 200, 400, 400, 400, 200, 200, 200, 400}
```
Comment (jpepin) [inline kubernetes/nodecollection_test.go:74]: The ordering in this list is starting to get unwieldy. We'll probably want to do something a little more readable the next time we add something, but that's a problem for _future us_

File: kubernetes/nodecollection.go [hunk 2]
```diff
@@ -86,6 +86,39 @@ func (cns ClientsetNodeSource) NodeAddress(node *v1.Node) (string, int32, error)
...
+func (m EndpointMask) SetAvailable(endpoint Endpoint, available bool) {
+	if available {
+		m[endpoint] = struct{}{}
+	} else {
+		delete(m, endpoint)
+	}
+}
+
+// GetAvailable gets the availability of an endpoint
+func (m EndpointMask) GetAvailable(endpoint Endpoint) bool {
```
Comment (jpepin) [inline kubernetes/nodecollection.go:117]: 🤠 We don't put Get in our Getter names around these here parts pardner 🌟 🌅

**Review by jpepin — COMMENTED** [review]
Looks great overall, just a few nits

File: kubernetes/nodecollection.go (hunk 2 above)
Comment (mnorbury) [reply kubernetes/nodecollection.go:117]: 🤠 👍

**Review by jpepin — APPROVED** [review]
Nice work 💯

**Review by jpepin — APPROVED** [review]

**Review by mollylogue — APPROVED** [review]

**Pull request merged**